Adds documentation for new OSB flags (#6403)
opensearch-project/documentation-website: _benchmark/reference/commands/command-flags.md

## limit
<!-- vale on -->

Limits the number of search results for recent test runs. Default is `10`.

<!-- vale off -->
## latency-percentiles
<!-- vale on -->

> Specifies a list of latency percentiles to report after the workload runs. Default is `50,90,99,99.9,99.99,100`. 

Reviewer: I believe latency percentiles and throughput percentiles only takes in numbers and not median, min, and max. To do median, min, and max, we'd have to specify `0,50,100`. Is this correct @peteralfonsi ?

> Specifies a list of latency percentiles to report after the workload runs. Default is `50,90,99,99.9,99.99,100`. 

Author: That's right, it only takes numeric percentiles. I will add a sentence about that.

<!-- vale off -->
## throughput-percentiles
<!-- vale on -->

> Specifies a list of throughput percentiles to report after the workload runs, in addition to mean/median/max/min. Default is `None`. 

Reviewer 2: Is this to be specified as floating point numbers?  Please elaborate here.

Reviewer: It can be taken as an int, float, or str. Based on [the code here](https://github.com/opensearch-project/opensearch-benchmark/blob/main/osbenchmark/metrics.py#L1333-L1337) that Peter added, it converts the user inputs to floats.

<!-- vale off -->
## randomization-enabled
<!-- vale on -->

> Enables randomization of values in range queries, where the values are drawn from standard value functions registered with `register_standard_value_source` in the workload's workload.py. Default is `False`. 

Reviewer: Nit: For folks who are not familiar, it would be good to add a small blurb defining "standard value functions".

Reviewer 2: What happens if there is no such function registered?  Please indicate that an error is raised if that is the case.

Reviewer: standard value functions can be defined as:
```
It's a function you call to generate a random pair of values for a certain field.
```
Source: https://github.com/opensearch-project/opensearch-benchmark/pull/455#issuecomment-1928366750

@peteralfonsi feel free to add more insight if needed. Govind has a good point, we should also address what happens if no function is registered.

Author: Yes, I will add a definition and the case when nothing is registered.

<!-- vale off -->
## randomization-repeat-frequency
<!-- vale on -->

> Sets what fraction of randomized query values can be repeated. Takes values between `0.0` and `1.0`. Default is `0.3`. Has no effect if `--randomization-enabled` is not used. 

Reviewer 3: ```suggestion
Sets what fraction of randomized query values can be repeated. Takes values between `0.0` and `1.0`. Default is `0.3`. This setting not work when `--randomization-enabled` is not used. 
```

<!-- vale off -->
## randomization-n
<!-- vale on -->

> Sets how many distinct repeatable pair values are generated for each operation when randomization is used. Default is `5000`. Has no effect if `--randomization-enabled` is not used. 

Reviewer 3: ```suggestion
Sets how many distinct repeatable pair values are generated for each operation when randomization is used. Default is `5000`. This setting does not work when `--randomization-enabled` is not used. 
```

## limit
<!-- vale on -->

Limits the number of search results for recent test runs. Default is `10`.

<!-- vale off -->
## latency-percentiles
<!-- vale on -->

> Specifies a comma-separated list of latency percentiles to report after the workload runs. Accepts ints or floats with values between `0` and `100` inclusive; does not accept `min`, `median`, `mean`, or `max`. Default is `50,90,99,99.9,99.99,100`. 

Reviewer 3: ```suggestion
Specifies a comma-separated list of latency percentiles to report after the workload runs. Accepts `ints` or `floats` with values between `0` and `100` inclusive. Does not accept `min`, `median`, `mean`, or `max`. Default is `50,90,99,99.9,99.99,100`. 
```
